akka-docs/src/main/paradox/scala/dispatchers.md

```diff
@@ -140,3 +163,254 @@ Note that it's not guaranteed that the *same* thread is used over time, since th
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder).
+
+The orange portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amount of turquoise (blocked, or sleeping as in our example) threads
+is very bad and leads to thread starvation.
+
+@@@ note
+
+If you own a Lightbend subscription you can use the commercial [Thread Starvation Detector](http://developer.lightbend.com/docs/akka-commercial-addons/current/starvation-detector.html)
+which will issue warning log statements if it detects any of your dispatchers suffering from starvation and other.
+It is a helpful first step to identify the problem is occurring in a production system,
+and then you can apply the proposed solutions as explained below.
+
+@@@
+
+![dispatcher-behaviour-on-bad-code.png](../images/dispatcher-behaviour-on-bad-code.png)
+
+In the above example we put the code under load by sending hundreds of messages to the blocking actor
+which causes threads of the default dispatcher to be blocked.
+The fork join pool based dispatcher in Akka then attempts to compensate for this blocking by adding more threads to the pool
+(`default-akka.actor.default-dispatcher 18,19,20,...`).
+This however is not able to help if those too will immediately get blocked,
+and eventually the blocking operations will dominate the entire dispatcher.
+
+In essence, the `Thread.sleep` operation has dominated all threads and caused anything
+executing on the default dispatcher to starve for resources (including any actor
+that you have not configured an explicit dispatcher for).
+
+@@@ note
+
+If you wrap up the blocking operation inside `scala.conccurrent.blocking` as follows:
+
+```scala
+Future {
+  scala.concurrent.blocking{
+    Thread.sleep(5000) //block for 5 seconds
+    println(s"Blocking future finished ${i}")
+  }
+}
+```
+
+the fork join pool can potentially generate a much larger number of compensating threads.
+This can possibly avoid the case where blocking threads dominate the default dispatcher,
+but there is a danger that the number of compensating threads can grow to the thread limit, and the memory is run out.
+
+Thus, the solution in the following section is more recommended and controlled approach.
+
+@@@
+
+## Solution: Dedicated dispatcher for blocking operations
+
+One of the most efficient methods of isolating the blocking behaviour such that it does not impact the rest of the system
+is to prepare and use a dedicated dispatcher for all those blocking operations.
+This technique is often referred to as as "bulk-heading" or simply "isolating blocking".
+
+In `application.conf`, the dispatcher dedicated to blocking behaviour should
+be configured as follows:
+
+```conf
+my-blocking-dispatcher {
+  type = Dispatcher
+  executor = "thread-pool-executor"
+  thread-pool-executor {
+    fixed-pool-size = 16
+  }
+  throughput = 100
```

Comment:
I don't think `throughput = 100` makes sense for blocking dispatcher, since blocking things typically take long time and then you want fairness, i.e. throughput = 1

I think we already have a snippet for a blocking dispatcher config, include same here